src/Symfony/Component/Form/Attribute/AsFormType.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Attribute;
+
+/**
+ * Register a model class (e.g. DTO, entity, model, etc...) as a FormType.
+ *
+ * @author Benjamin Georgeault <git@wedgesama.fr>
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class AsFormType
```

Comment:
I don't like the name `AsFormType` as it looks like it will register the service as a form type (tagging it as `form.type`), while it has a totally different behavior (it defines that class as providing metadata to configure a form type, not as _being_ a form type)

Reply:
What do you think of:
1. `AsFormMetadata`
2. `FormMetadata`
3. `AsForm`

I have a preference for the 1.